src/OutfitMakerSecondary.java

```diff
@@ -0,0 +1,89 @@
+package components.OutfitMaker;
+import java.util.Random;
+import java.util.Set;
+import components.sequence.Sequence;
+import components.sequence.Sequence1L;
+import components.set.Set;
+import components.set.Set1L;
+
+
+public abstract class OutfitMakerSecondary implements OutfitMaker {
+
+    @Override
+    public final Set<String> getShoes() {
```

Comment:
Based on the kernels, I think you implemented the methods backward. You should implement your secondary methods here (i.e., hasItem and getOutfit). Or, you might need to rework your layers.